### PR TITLE
Public private subnet option

### PR DESCRIPTION
**Pull request by limed**
This allows us to drop ec2 instance into the public or private ec2 subnet groups

### CODE REVIEW

File: ec2.template [hunk 1]
```diff
@@ -134,6 +143,26 @@
...
+    },
+    "SubnetLocationPublic": {
+      "Fn::Equals": [
+        {
+          "Ref": "SubnetLocation"
+        },
+        "public"
+      ]
+    },
+    "NotSubnetLocationPublic": {
```
Comment (gozer) [inline ec2.template]: For the sake of logical clarity, wouldn't naming this SubnetLocationPrivate be more sensical?

File: ec2.template [hunk 2]
```diff
@@ -472,10 +605,24 @@
         ]
       }
     },
+    "AutoScalingGroupPublic": {
```
Comment (gozer) [inline ec2.template:616]: AutoScalingGroupPublic / AutoScalingGroup outputs isn't ideal to me, but as discussed, CF doesn't let you do much in the output section, so if it's impossible to have conditionnal outputs with the same exact name, this is good enough backwards compatibility to me.

**Comment by gozer** [comment]
The ec2.README should be updated to reflect this option

File: ec2.template [hunk 3]
```diff
@@ -354,8 +383,112 @@
         }
       }
     },
-    "AutoScalingGroup": {
+    "AutoScalingGroupPublic": {
+      "Type": "AWS::AutoScaling::AutoScalingGroup",
+      "Condition": "SubnetLocationPublic",
+      "AssociatePublicIpAddress": "true",
```
Comment (limed) [inline ec2.template]: So just realized that this is the wrong place to put it in, this particular option only works in `AWS::AutoScaling::LaunchConfiguration` will do some testing and fix up this PR to reflect that

**Comment by gozer** [comment]
@limed can we please make extra sure we have tested (by deploying them) CF stacks templates before opening a PR? We don't have great testing in place for this stuff yet, so at a minimum, ensuring the template has deployed fine is a minimum. Thanks!

**Pull request merged**